lib/licensed/sources/nuget.rb

```diff
@@ -188,16 +189,27 @@ def enumerate_dependencies
             # Already know this package from another target
             next if dependencies.key?(id)
 
-            path = File.join(nuget_packages_dir, json["libraries"][reference_key]["path"])
-            dependencies[id] = NuGetDependency.new(
-              name: id,
-              version: version,
-              path: path,
-              metadata: {
-                "type" => NuGet.type,
-                "name" => name
-              }
-            )
+            # Find package in one of nuget folders
+            nuget_packages_dirs.each do |packages_dir|
```

Comment:
the nested looping bumps up the cyclomatic complexity of the function quite a bit - can you move the `path` construction logic to a helper method that finds/constructs the path passed into the dependency intializer?